### DISCUSSION TITLE
Large index.css module

### DISCUSSION THREAD

**Discussion by FlatMapIO** [chakra-ui/panda#2236] [Q&A] [answered]
When I use park-ui + panda in solid-start, the style change will cause the browser to reload ~6m index.css module, which makes HMR meaningless. When viewing this file, it is basically a huge list of tokens that will never change Is there a way to optimize this?

**Comment by astahmer** [comment]
yes, there are a few things that you can do to improve the overall performance:

- [reduce your overall tokens count by omitting those that you know you will never use](https://panda-css.com/docs/concepts/hooks#tweak-the-config)

- [switch to CLI instead of postcss plugin](https://panda-css.com/docs/guides/debugging#the-postcss-plugin-sometimes-seems-slow-or-runs-too-frequently)
if the issue persists after swapping to using the CLI, a performance profiling could help us debug your issues
https://panda-css.com/docs/guides/debugging#performance-profiling

- [remove unused CSS vars using a postcss plugin](https://panda-css.com/docs/concepts/hooks#remove-unused-variables-from-final-css)

**Comment by FlatMapIO** [comment]
After testing, the perfect solution is to use turbo + cli, Thank you very much for your guidance @astahmer!

**Comment by FlatMapIO** [comment]
@astahmer There seems to be a bug in the cli watch mode? when I keep adding blank lines to the code, `style.css` keeps growing until it's huge, and when I look at the content, it seems to be all duplicate styles. The only way to do this at the moment is to manually run cssgen after coding for a while

**Reply by astahmer** [reply]
which version of Panda are you using ?
could you provide a minimal reproduction ?

**Reply by FlatMapIO** [reply]
package.json
```json
{
  "scripts": {
   "dev:css": "DEV=true pnpm css:gen -w",
 }
 "devDependencies": {
   "@pandacss/dev": "^0.32.0",
   "@park-ui/panda-preset": "^0.35.2",
 }
}
```

panda.config.ts

```tsx
import { defineConfig } from '@pandacss/dev'
import { createPreset } from '@park-ui/panda-preset'
import {removeUnusedCssVars} from './scripts/postcss/remove-unused-css-vars'
import {removeUnusedKeyframes} from './scripts/postcss/remove-unused-keyframes'

export default defineConfig({
	// Whether to use css reset
	preflight: true,
	jsxFramework: 'solid',
	jsxStyleProps: 'minimal',
	// lightningcss: true,
	// Where to look for your css declarations
	include: ['./src/**/*.{ts,tsx}'],
	presets: [
		'@pandacss/preset-base',
		createPreset({
			accentColor: 'neutral',
			grayColor: 'neutral',
			// additionalColors: ['red'],
		}),
	],
	// Files to exclude
	exclude: [],

	// Useful for theme customization
	theme: {
		extend: {},
	},

	// Optimize options: https://github.com/chakra-ui/panda/discussions/2236
	hooks: {
		'cssgen:done': ({ artifact, content }) => {
			if (artifact === 'styles.css') {
				return removeUnusedCssVars(removeUnusedKeyframes(content))
			}
		},
	},
	// The output directory for your css system
	outdir: 'styled-system',
})

```

**Reply by astahmer** [reply]
I cannot reproduce the issue with this
![image](https://github.com/chakra-ui/panda/assets/47224540/77ee2e1a-9f41-4068-9adc-1cff85083e11)

please provide a github repo where you can replicate the issue so I can help 🙏

**Reply by astahmer** [reply]
you can try using this as a base
https://github.com/astahmer/panda-vite-template/tree/repro/large-index-2236

**Reply by FlatMapIO** [reply]
I found the problem when I was about to reproduce it, I noticed that my `panda cssgen` command had the 
`--lightningcss` parameter in it, and after removing it everything worked fine. It looks like this is a bug caused by lightningcss integration

**Reply by astahmer** [reply]
I just tried on that repo/branch I sent earlier with `pnpm panda cssgen --lightningcss -w` and still couldnt reproduce 🫤

**Reply by FlatMapIO** [reply]
It's a little hard to understand 🤣, I'll continue working on the reproduce repo later.

**Reply by FlatMapIO** [reply]
After adding park-ui, the problem has been reproduced.
https://github.com/astahmer/panda-vite-template/pull/1

**Reply by FlatMapIO** [reply]
`removeUnusedCssVars(removeUnusedKeyframes(content))` will slow down the css growth, but it will still grow. If remove is not configured, it will almost double every time you compile.

**Reply by astahmer** [reply, marked as answer]
thank you for the minimal reproduction, this was super helpful to debug 🙏 
should be fixed with https://github.com/chakra-ui/panda/pull/2249
